### ISSUE TITLE
Handling of nulls when applying defaults

### ISSUE THREAD

**Issue by gtuccini** [justinrainbow/json-schema#377]
Currently null default values are not correctly applied. I think this happens because validateCommonProperties, in UndefinedConstraint.php, uses isset to check whether a default value has been defined and it cannot distinguish between a null default value and an undefined one.

```php
// e.g.
if (!$this->getTypeCheck()->propertyExists($value, $i) && isset($propertyDefinition->default)) {
```

Moreover, when the subject is an array and default properties or items have been defined, non-null default values are applied even if the properties or the items have a null value. Again, I think this happens because the code uses isset to check whether the property or the item exists.

```php
if (!isset($value[$i]) && isset($propertyDefinition->default)) {
// ...
if (!isset($value[$i]) && isset($itemDefinition->default)) {
```

**Comment by erayd** [comment]
@gtuccini Thanks for reporting this - will see if I can get this sorted over the weekend.

**Comment by erayd** [comment]
Fixed in c92dfae.

**Comment by gtuccini** [comment]
That was fast! Thank you very much.

**Comment by erayd** [comment]
You're most welcome :-).

**Comment by erayd** [comment]
@bighappyface Can this issue be closed? The fix is merged in both [6.0.0-dev](https://github.com/justinrainbow/json-schema/tree/6.0.0-dev) and [5.2.0](https://github.com/justinrainbow/json-schema/tree/5.2.0).
